lib/dal/src/migrations/U2600__attribute_value_dependency_graph_summary.sql

```diff
@@ -0,0 +1,1489 @@
+                    ON avbtap.object_id = av.id
+            WHERE avbtap.belongs_to_id = this_new_row.id
+        LOOP
+            PERFORM attribute_value_dependencies_update_v1(
+                this_tenancy ->> 'tenancy_workspace_pk',
+                this_visibility ->> 'visibility_change_set_pk',
+                this_visibility ->> 'visibility_deleted_at',
+                this_attribute_value.id
+            );
+        END LOOP;
+    END IF;
+
+    RAISE DEBUG 'attribute_prototype_argument_create_v1: Created AttributePrototypeArgument(%)', this_new_row;
+
+    object := row_to_json(this_new_row);
+END;
+$$ LANGUAGE PLPGSQL VOLATILE;
+
+CREATE OR REPLACE FUNCTION belongs_to_table_create_v1(this_table_name text,
+                                                      this_object_table text,
+                                                      this_belongs_to_table text)
+    RETURNS VOID
+AS
+$$
+DECLARE
+    create_table text;
+BEGIN
+    create_table := format('CREATE TABLE %1$I ( '
+                           ' pk                          ident primary key default ident_create_v1(), '
+                           ' id                          ident not null default ident_create_v1(), '
+                           ' object_id                   ident                   NOT NULL, '
+                           ' belongs_to_id               ident                   NOT NULL, '
+                           ' tenancy_workspace_pk        ident, '
+                           ' visibility_change_set_pk    ident                   NOT NULL DEFAULT ident_nil_v1(), '
+                           ' visibility_deleted_at       timestamp with time zone, '
+                           ' created_at                  timestamp with time zone NOT NULL DEFAULT CLOCK_TIMESTAMP(), '
+                           ' updated_at                  timestamp with time zone NOT NULL DEFAULT CLOCK_TIMESTAMP() '
+                           '); '
+                           'CREATE UNIQUE INDEX %1$s_visibility_tenancy ON %1$I (id, '
+                           '                                    tenancy_workspace_pk, '
+                           '                                    visibility_change_set_pk); '
+                           'ALTER TABLE %1$I '
+                           '    ADD CONSTRAINT %1$s_object_id_is_valid '
+                           '        CHECK (check_id_in_table_v1(%2$L, object_id)); '
+                           'ALTER TABLE %1$I '
+                           '    ADD CONSTRAINT %1$s_belongs_to_id_is_valid '
+                           '        CHECK (check_id_in_table_v1(%3$L, belongs_to_id)); '
+                           'CREATE UNIQUE INDEX %1$s_single_association ON %1$I (object_id, '
+                           '                                        tenancy_workspace_pk, '
+                           '                                        visibility_change_set_pk) '
+                           '    WHERE visibility_deleted_at IS NULL; '
```

Comment:
The old one wasn't taking into account that we only want **undeleted** records to count for the uniqueness.

Reply:
Haven't read all of it yet, but does this mean we're not getting attribute values for deleted components at all?

Reply:
we still show soft deleted components that  exist on head inside of the changeset as the ghosted components, so you can still look at their attribute editor. So we still need those values in the summary

Reply:
nevermind, we don't show the attribute editro for the ghosted components